src/docs/sdk/performance/ui-event-transactions.mdx

```diff
@@ -0,0 +1,160 @@
+    Given an ongoing UI event transaction
+    When the user triggers the same UI element with a different event
+    Or the user triggers a different UI element
+    Then the SDK finishes the ongoing transaction
+    And sets the status to OK
+    And waits for the children to finish
+    And cancels the timeout
+    And starts a new transaction
+```
+
+## Binding to scope
+
+On platforms mainly interacting with the static API, such as mobile, it's typical to bind auto-generated
+transactions to the scope so that users can access them via the static API. We recommend binding UI event
+transactions to the scope on these platforms. The following extra rules apply as UI event transactions
+could interfere with other auto-generated transactions.
+
+```Gherkin
+Scenario: Same UI element with different event
+    Given an ongoing UI event transaction
+    When the user triggers the same UI element with a different event
+    Or the user triggers a different UI element
+    Then the SDK finishes the ongoing transaction
+    And sets the status to OK
+    And waits for the children to finish
+    And cancels the timeout
+    And removes the ongoing transaction from the scope
+    And starts a new transaction
+    And puts the new transaction on the scope
+
+Scenario: UI event triggered but transaction ended
+    Given an auto-generated transaction from a any UI event
+    And the transaction already  is finished
+    When the user triggers the same UI event
+    Then the SDK starts a new UI event transaction
+
+Scenario: Manually created transaction bound to the scope
+    Given an ongoing manually created transaction by the user bound to the
+        scope
+    When the SDK creates a new UI event transaction
+    Then the SDK doesn't add the UI event transaction to the scope 
+```
+
+
+### Screen load transactions
+
+This section deals specifically with auto-generated transactions for loading screens. If your SDK has
+other types of auto-generated transactions, please update the specification here.
+
+```Gherkin
+Scenario: Ongoing UI event transaction
+    Given an ongoing UI event transaction
+    When the SDK creates a new screen load transaction
+    Then the SDK finishes the ongoing UI event transaction
+    And removes it from the scope
+    And sets the status to canceled
+    And waits for its children to finish
+
+Scenario: Ongoing screen load transaction
+    Given an ongoing screen load transaction
+    When the SDK starts a new UI event transaction
+    Then the SDK doesn't bind the new UI event transaction to the scope
+```
+
+## Transaction name
+
+The user should be able to identify to which UI element the UI event transaction belongs on which screen
+by only looking at the transaction name. Choose whatever works best for your specific platform. Be
+attentive not to use any PII in the transaction name. One recommendation is 
+`screen name + view identifier || accessibility identifier`. On Android, this would map to
```

Comment:
`accessibility identifier` may have PII because of screen readers, do we guard this against `sendDefaultPii`?

Reply:
For context, accessibility id was a problem for https://docs.sentry.io/platforms/react-native/touchevents/ and we needed to provide a way out, so it can't be used by default.

Reply:
we only use the view identifier

Reply:
@philipphofmann so I'd rather remove the `accessibility identifier` recommendation or guard it against `sendDefaultPii ` in the spec.
Also, you don't always have an identifier, in this case, we should write in the spec that you don't start a transaction at all.

Reply:
yep, I agree we should mention that - we don't start any transactions if we're unable to find the view id. Can be improved, but usually the views that trigger action always have some ids on them to attach listeners to them.

Reply:
@romtsn depends on the platform, it's not that common having an `id` (key) in Flutter unless people use it for UI tests or something because you set the listeners directly in the declaration.

Reply:
@marandaneto, on Cocoa, we use the method the UI element calls. I added this now. I also mention that `Choose whatever works best for your specific platform.`. Do you think that's fine or should I add more information, @marandaneto?

Reply:
@philipphofmann https://github.com/getsentry/sentry-cocoa/blob/efeb69e9c50860c600d68640b1f465cc83a1a7f7/Sources/Sentry/SentryUIEventTracker.m#L130 and https://github.com/getsentry/sentry-cocoa/blob/d9757459a4a288ad57b2a8348def9eb839cbe72b/Sources/Sentry/SentryBreadcrumbTracker.m#L219
This can be PII and isn't guarded with `defaultPii`

Reply:
> @marandaneto, on Cocoa, we use the method the UI element calls. I added this now. I also mention that `Choose whatever works best for your specific platform.`. Do you think that's fine or should I add more information, @marandaneto?

Yes, all good for the spec, but I guess we should raise an issue to guard the linked code against `sendDefaultPii` or remove it altogether, what do you think?

Reply:
I will raise issues, yes. 🥇 for finding them.

Reply:
https://github.com/getsentry/sentry-cocoa/issues/2406